### PR TITLE
DB-6346 allow more scenarios to qualify for CSSQ flattening

### PR DESCRIPTION
**Pull request by yxia92**
Please see [DB-6346](https://splicemachine.atlassian.net/browse/DB-6346) for the fix description.

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java [hunk 1]
```diff
@@ -129,47 +129,54 @@ public boolean hasNext() {
...
-
-                if (leftSideIterator.hasNext()) {
-                    // If we haven't consumed left side iterator, submit a task to scan righ side
-                    ExecRow execRow = leftSideIterator.next();
-                    GetNLJoinIterator getNLJoinIterator = GetNLJoinIterator.makeGetNLJoinIterator(joinType,
-                            operationContextList.remove(0), execRow.getClone());
-                    completionService.submit(getNLJoinIterator);
-                    nLeftRows++;
-                }
+            while (true) {
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java]: Why is this loop needed?

Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java]: When we get a match from the SSQ, I want to skip the remaining matching rows, and let  rightSideNJIterator iterate to the end.

Comment (dgomezferro) [reply splice_machine/src/main/java/com/splicemachine/derby/stream/function/NLJoinFunction.java]: I see it now, thanks!

**Comment by OlegMazurov** [comment]
Please, resolve conflicts.

**Comment by yxia92** [comment]
Updated the PR by rebasing to latest.

**Comment by wtang16** [comment]
Can we backport this to 2.6.1? Customer requires this version.

**Pull request merged**